File: ADS_2_POO/ADS2/CNPJ.cs

public class CNPJ : Documentos
{

    public CNPJ(string numeroInput) : base(numeroInput, 14, 12)
    {
        pesoUm = new int[] { 5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2 };
        pesoFinal = new int[] { 6, 5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2 };
        numero = DocumentosFormatacao.FormataNumero(numero, tamanhoComDigitos, tamanhoSemDigitos);
        numeroCorreto = CalcularNumeroCorreto(numero);
        mascaraNumero = numeroCorreto.Substring(0, 2) + "." + numeroCorreto.Substring(2, 3) + "." + numeroCorreto.Substring(5, 3) + "/" + numeroCorreto.Substring(8, 4) + "-" + numeroCorreto.Substring(12, 2);
    }

}

File: ADS_2_POO/Python Version/cnpj.py
from documentosFormatacao import DocumentosFormatacao
from documentos import Documentos

class CNPJ(Documentos):
    def __init__(self, numero_input):
        super().__init__(numero_input, 14, 12)
        self.peso_um = [5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2]
        self.peso_final = [6, 5, 4, 3, 2, 9, 8, 7, 6, 5, 4, 3, 2]
        self.numero = DocumentosFormatacao.formata_numero(numero_input, self.tamanho_com_digitos, self.tamanho_sem_digitos)
        self.numero_correto = self.calcular_numero_correto(self.numero)
        self.mascara_numero = f"{self.numero_correto[:2]}.{self.numero_correto[2:5]}.{self.numero_correto[5:8]}/{self.numero_correto[8:12]}-{self.numero_correto[12:]}"

# Se você não tiver uma classe DocumentosFormatacao e a função formata_numero, você precisará implementá-las ou adaptar o código conforme necessário.
